app/javascript/App.jsx

```diff
@@ -11,10 +12,15 @@ export default function App() {
   const currentUser = useAuth();
   const pageHeight = currentUser?.signed_in ? 'regular-height' : 'no-header-height';
 
+  // //i18n
+  const { i18n } = useTranslation();
+  useEffect(() => {
+    i18n.changeLanguage(currentUser?.language);
+  }, [currentUser]);
```

Comment:
Shouldn't this be `currentUser?.language'? That way, it only changes if the user's language changes?